File: JavaScript/Week-02/Day-2/MatchMaking.ts
'use strict';
// Write a function that joins two array by matching one girl with one boy in a new array
// If someone has no pair, he/she should be the element of the array too
// Exepected output: ["Eve", "Joe", "Ashley", "Fred"...]

let girls: string[] = ['Eve', 'Ashley', 'Claire', 'Kat', 'Jane','Alice'];
let boys: string[] = ['Joe', 'Fred', 'Tom', 'Todd', 'Neef', 'Jeff'];

function makingMatches(array1: string[], array2: string[]) {
  let length1: number = array1.length;
  let length2: number = array2.length;
  let finalArray: String[] = [];
  if (length1 === length2) {
    for (let x: number = 0; x < length1; x++) {
      finalArray.push(array1[x], array2[x]);
    }
    return finalArray;
  } if (length1 > length2) {
    for (let x: number = 0; x < length2; x++) {
      finalArray.push(array1[x], array2[x]);
    }
    for (let x: number = length2; x < length1; x++) {
      finalArray.push(array1[x]);
    }
    return finalArray;
  } else {
    for (let x: number = 0; x < length1; x++) {
      finalArray.push(array1[x], array2[x]);
    }
    for (let x: number = length1; x < length2; x++) {
      finalArray.push(array2[x]);
    }
    return finalArray;
  }
}
console.log(makingMatches(girls, boys));

File: Java/Week2/Day2/Matchmaking.java
import java.util.*;

public class Matchmaking {
  public static void main(String... args) {
    ArrayList<String> girls = new ArrayList<String>(Arrays.asList("Eve", "Ashley", "Claire", "Kat", "Jane"));
    ArrayList<String> boys = new ArrayList<String>(Arrays.asList("Joe", "Fred", "Tom", "Todd", "Neef", "Jeff"));

    // Write a method that joins the two lists by matching one girl with one boy
    // into a new list
    // If someone has no pair, he/she should be the element of the list too
    // Exepected output: "Eve", "Joe", "Ashley", "Fred"...

    System.out.println(makingMatches(girls, boys));
  }

  public static ArrayList makingMatches(ArrayList<String> Girls, ArrayList<String> Boys) {
    ArrayList<String> paired = new ArrayList<>();
    if (Girls.size() > Boys.size()) {
      for (int i = 0; i < Boys.size(); i++) {
        paired.add(Girls.get(i));
        paired.add(Boys.get(i));
      }
      paired.add(Girls.get(Boys.size() - 1));
      paired.add("No boys");
    } else if (Girls.size() < Boys.size()) {
      for (int i = 0; i < Girls.size(); i++) {
        paired.add(Girls.get(i));
        paired.add(Boys.get(i));
      }
      paired.add("No girl");
      paired.add(Boys.get(Girls.size() - 1));
    } else {
      for (int i = 0; i < Girls.size(); i++) {
        paired.add(Girls.get(i));
        paired.add(Boys.get(i));data1
      }
    }
    return paired;
  }
}